src/app/cli/src/commands/system_diagnose_command.rs

```diff
@@ -0,0 +1,325 @@
+            )?;
+        }
+        Ok(())
+    }
+}
+
+///////////////////////////////////////////////////////////////////////////////
+
+#[derive(thiserror::Error, Debug)]
+enum DiagCheckError {
+    #[error(transparent)]
+    Failed(#[from] CommandExecError),
+    #[error(transparent)]
+    Internal(
+        #[from]
+        #[backtrace]
+        InternalError,
+    ),
+}
+
+#[derive(Error, Debug)]
+#[error(
+    "{message} \n {error_log}
+    Make sure to follow kamu installation instructions to correctly configure \
+     docker or podman:
+        https://docs.kamu.dev/cli/get-started/installation/"
+)]
+pub struct CommandExecError {
+    pub message: String,
+    pub error_log: String,
+}
+
+impl From<std::io::Error> for DiagCheckError {
+    fn from(e: std::io::Error) -> Self {
+        Self::Failed(CommandExecError {
+            message: "Unable to perform io operation".to_string(),
+            error_log: e.to_string(),
+        })
+    }
+}
+
+///////////////////////////////////////////////////////////////////////////////
+
+#[async_trait::async_trait]
+trait DiagCheck {
+    fn name(&self) -> String;
+    async fn run(&self) -> Result<(), DiagCheckError>;
+}
+
+///////////////////////////////////////////////////////////////////////////////
+
+struct CheckContainerRuntimeIsInstalled {
+    container_runtime: Arc<ContainerRuntime>,
+}
+
+#[async_trait::async_trait]
+impl DiagCheck for CheckContainerRuntimeIsInstalled {
+    fn name(&self) -> String {
+        format!("{} installed", self.container_runtime.config.runtime)
+    }
+    async fn run(&self) -> Result<(), DiagCheckError> {
+        let command_res = self.container_runtime.info().output().await.int_err()?;
+        handle_output_result(command_res)
+    }
+}
+
+///////////////////////////////////////////////////////////////////////////////
+
+struct CheckContainerRuntimeImagePull {
+    container_runtime: Arc<ContainerRuntime>,
+}
+
+#[async_trait::async_trait]
+impl DiagCheck for CheckContainerRuntimeImagePull {
+    fn name(&self) -> String {
+        format!("{} can pull images", self.container_runtime.config.runtime)
+    }
+    async fn run(&self) -> Result<(), DiagCheckError> {
+        self.container_runtime
+            .pull_image(BUSYBOX, None)
+            .await
+            .int_err()?;
+
+        Ok(())
+    }
+}
+
+///////////////////////////////////////////////////////////////////////////////
+
+struct CheckContainerRuntimeRootlessRun {
+    container_runtime: Arc<ContainerRuntime>,
+}
+
+#[async_trait::async_trait]
+impl DiagCheck for CheckContainerRuntimeRootlessRun {
+    fn name(&self) -> String {
+        format!(
+            "{} rootless run check",
+            self.container_runtime.config.runtime
+        )
+    }
+    async fn run(&self) -> Result<(), DiagCheckError> {
+        let run_args = RunArgs {
+            image: BUSYBOX.to_string(),
+            container_name: Some("check-rootless-run".to_owned()),
```

Comment:
Let's prefix the name with `kamu-` so that user knew what this container is about, and also add a random suffix, i.e. `kamu-check-rootless-af4s6` - this way if somehow we end up running two of these checks in parallel (e.g. during tests) they wouldn't conlfict.